transformer/external/executabletransformer.go

```diff
@@ -32,20 +32,50 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// IOMode is the type of the mode used for I/O of detect and transform scripts.
+type IOMode string
+
+const (
+	envMode      IOMode = "env"
+	argMode      IOMode = "arg"
+	envDelimiter        = "="
+)
+
 // Executable implements transformer interface and is used to write simple external transformers
 type Executable struct {
 	Config     transformertypes.Transformer
 	Env        *environment.Environment
 	ExecConfig *ExecutableYamlConfig
 }
 
+// InputFormat input/output mode definition
+type InputFormat struct {
+	Mode    IOMode `yaml:"mode"`
+	EnvName string `yaml:"envName,omitempty`
+	Path    string
+}
+
+// OutputFormat input/output mode definition
+type OutputFormat struct {
+	Mode    IOMode `yaml:"mode"`
+	EnvName string `yaml:"envName,omitempty`
+	Path    string
+}
+
+// CommandFormat input/output mode and command definitions for detect and transform scripts
+type CommandFormat struct {
+	Cmd    environmenttypes.Command `yaml:"cmd"`
+	Input  InputFormat              `yaml:"input"`
+	Output OutputFormat             `yaml:"output"`
+}
+
 // ExecutableYamlConfig is the format of executable yaml config
 type ExecutableYamlConfig struct {
 	EnableQA           bool                       `yaml:"enableQA"`
 	Platforms          []string                   `yaml:"platforms"`
-	DirectoryDetectCMD environmenttypes.Command   `yaml:"directoryDetectCMD"`
-	TransformCMD       environmenttypes.Command   `yaml:"transformCMD"`
 	Container          environmenttypes.Container `yaml:"container,omitempty"`
+	DetectCmdFormat    CommandFormat              `yaml:"directoryDetect"`
```

Comment:
Lets also think through (need not implement grpc in this PR) how the struct should look like if we have to support grpc and design it.

Reply:
Will be covered in follow-up PR as discussed.